src/main.rs

```diff
@@ -131,10 +131,15 @@ fn main() {
     // Create a channel for server to communicate with network
     let (tx, rx) = mpsc::unbounded();
 
-    let lan_discovery_sender = LanDiscoverySender::new(tx.clone(), dht_pk, udp_addr.is_ipv6());
+    let lan_discovery_future = if cli_config.lan_discovery_enabled {
+        LanDiscoverySender::new(tx.clone(), dht_pk, udp_addr.is_ipv6()).run()
+    } else {
+        Box::new(future::empty())
```

Comment:
You combine futures like this: `future.select(lan_discovery_future)`. Shouldn't `future::empty()` resolve as fast as possible?

Reply:
Empty is a future that never resolves.

Reply:
Thanks.